packages/plugin-dev/src/browser/hosted-plugin-manager-client.ts

```diff
@@ -306,8 +323,52 @@ export class HostedPluginManagerClient {
         }
     }
 
+    protected async launchVSCode({ arguments: { args } }: LaunchVSCodeRequest): Promise<LaunchVSCodeResult> {
+        let result = {};
+        let instanceURI;
+
+        const sessions = this.debugSessionManager.sessions.filter(session => session.id !== this.connection.sessionId);
+
+        if (sessions.length > 0) {
+            sessions.forEach(session => session.terminate(false));
+            await this.startDebugSessionManager();
+            instanceURI = this.pluginInstanceURL;
+        } else {
+            instanceURI = await this.debug(this.getDebugPluginConfig(args));
+        }
+
+        if (instanceURI) {
+            const instanceURL = new URL(instanceURI);
+            if (instanceURL.port) {
+                result = Object.assign(result, { rendererDebugPort: instanceURL.port });
+            }
+        }
+        return result;
+    }
+
     protected getErrorMessage(error: Error): string {
-        return error.message.substring(error.message.indexOf(':') + 1);
+        return error?.message?.substring(error.message.indexOf(':') + 1) || '';
+    }
```

Comment:
Same as the previous comment.

Reply:
https://github.com/eclipse-theia/theia/pull/8706#discussion_r590097344

Reply:
Fixed